### ISSUE TITLE
Build failed with error saying annotation processors must be explicity declared now

### ISSUE THREAD

**Issue by volsahin** [arturbosch/detekt#2590]
Hi there,
I recently added these two dependencies to write some custom detekt rules.

```
implementation 'io.gitlab.arturbosch.detekt:detekt-api:1.7.4'
testImplementation 'io.gitlab.arturbosch.detekt:detekt-test:1.7.4'
```
But build failed with an exception saying:

**Annotation processors must be explicitly declared now. The following dependencies on the compile classpath are found to contain annotation processor. Please add them to the annotationProcessor configuration.
(org.jetbrains.kotlin:kotlin-compiler-embeddable:1.3.71)**

Adding this kotlin-compiler-embeddable dependency with kapt didn't help.
What should i need to do for building project with these dependencies ?

Thanks

**Comment by arturbosch** [comment]
Hi,
do you know who prints this warning?
Which plugins do you use?
It seems that you apply some plugin to your `custom-detekt-rules` module which shouldn't be necessary for detekt.

**Comment by volsahin** [comment]
Thanks for response.
I've seperated module for custom-detekt-rules and basicly have one custom-rule class and one class for testing this rule class. So i added the dependencies below.
In this new custom-detekt-rules module that i create i have only basic plugins like
`com.android.library`,
`kotlin-android`, 
`kotlin-android-extensions`. I haven't add any extra plugins.
I think detekt uses kotlin-compiler-embeddable and detekt has some kind of annotation processor that i also need to add with kapt. I can go with `includeCompileClasspath = true` but since it's deprecated and doesn't solve the real problem, i try to find what is the main problem

**Comment by volsahin** [comment]
Ok problem solved with addition of
`annotationProcessor "org.jetbrains.kotlin:kotlin-compiler-embeddable:1.3.71"`
I didn't get why kapt didn't work but it seems this line solves the issue
